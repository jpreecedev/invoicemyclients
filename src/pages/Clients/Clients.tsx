import React from 'react'
import { Link } from 'react-router-dom'
import { Plus } from 'react-feather'

import { AuthenticatedPage } from '../../components/AuthenticatedPage'
import { ROUTES } from '../../routes'

const ClientsPage = () => {
  return (
    <AuthenticatedPage title="Clients">
      <Link to={ROUTES.clients.add} className="btn btn-primary d-inline-flex">
        <Plus className="mr-2" />
        Add New Client
      </Link>
    </AuthenticatedPage>
  )
}

export { ClientsPage }
