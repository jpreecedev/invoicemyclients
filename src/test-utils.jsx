import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
