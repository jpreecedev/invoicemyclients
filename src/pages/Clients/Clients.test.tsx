import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { ClientsPage } from './Clients'

describe('<Clients /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<ClientsPage />).container).toMatchSnapshot()
  })
})
