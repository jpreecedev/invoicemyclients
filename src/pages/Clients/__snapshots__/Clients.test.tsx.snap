// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clients /> tests should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        InvoiceMyClients.com
      </a>
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            class="fe fe-x"
          />
        </button>
        <ul
          class="navbar-nav ml-auto"
        >
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/dashboard"
            >
              Dashboard
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/clients"
            >
              Clients
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/invoices"
            >
              Invoices
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/nagging"
            >
              Nagging
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              class="nav-link"
              href="/support"
            >
              Support
            </a>
          </li>
        </ul>
        <a
          class="navbar-btn btn btn-sm btn-primary lift ml-auto"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <main
    class="mt-5 mb-10 pb-10"
  >
    <h1
      class="container"
    >
      <div
        class="row"
      >
        <h1>
          Clients
        </h1>
      </div>
      <div
        class="row"
      >
        <a
          class="btn btn-primary d-inline-flex"
          href="/clients/add"
        >
          <svg
            class="mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
          Add New Client
        </a>
      </div>
    </h1>
  </main>
</div>
`;
